Gems/WhiteBox/Code/Source/SubComponentModes/EditorWhiteBoxTransformMode.cpp

```diff
@@ -602,9 +608,9 @@ namespace WhiteBox
             {
                 const AZ::Vector3 vertexLocalPosition =
                     (transformSelection->m_vertexPositions[vertexIndex++] - transformSelection->m_localPosition);
-                const AZ::Vector3 vertexPosition =
-                    (vertexLocalPosition * (AZ::Vector3::CreateOne() + (action.m_start.m_sign * action.LocalScaleOffset()))) +
-                    transformSelection->m_localPosition;
+                const AZ::Vector3 manipulatorScale = (AZ::Vector3::CreateOne() + (action.m_start.m_sign * 
+                    (scaleType == ScaleType::Uniform ? AZ::Vector3(action.LocalScaleOffset().GetZ()) : action.LocalScaleOffset()))); 
```

Comment:
suggestion: To save squashing this on to a line with the ternary operator maybe we could pull out the scale part to a separate lambda to make it a bit clearer?

```c++
const AZ::Vector3 scale = [&action]
{
    if (scaleType == ScaleType::Uniform)
    {
        return AZ::Vector3(action.LocalScaleOffset().GetZ());
    }

    if (scaleType == ScaleType::NonUniform)
    {
        return action.LocalScaleOffset();
    } 

    return AZ::Vector3::CreateOne();
}();

const AZ::Vector3 manipulatorScale = AZ::Vector3::CreateOne() + (action.m_start.m_sign * scale); 
```